Modules/Office365DSC/DSCResources/MSFT_TeamsGuestSettings/MSFT_TeamsGuestSettings.psm1

```diff
@@ -0,0 +1,183 @@
+        GlobalAdminAccount        = $GlobalAdminAccount
+    }
+
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $TeamName,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowCreateUpdateChannels,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowDeleteChannels,
+
+
+        [Parameter()]
+        [ValidateSet("Present", "Absent")]
+        [System.String]
+        $Ensure = "Present",
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    if ('Absent' -eq $Ensure)
+    {
+        throw "This resource cannot delete Managed Properties. Please make sure you set its Ensure value to Present."
+    }
+
+    Test-TeamsServiceConnection -GlobalAdminAccount $GlobalAdminAccount
+
+    $team = Get-Team |  Where-Object {$_.DisplayName -eq $TeamName}
+    if ($null -eq $team)
+    {
+        throw "Team with Name $TeamName doesnt exist in tenant"
+    }
+
+    $CurrentParameters = $PSBoundParameters
+    $CurrentParameters.Remove("TeamName")
+    $CurrentParameters.Add("GroupId", $team.GroupId)
+    $CurrentParameters.Remove("GlobalAdminAccount")
+    $CurrentParameters.Remove("Ensure")
+    Set-TeamGuestSettings @CurrentParameters
+}
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $TeamName,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowCreateUpdateChannels,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowDeleteChannels,
+
+
+        [Parameter()]
+        [ValidateSet("Present", "Absent")]
+        [System.String]
+        $Ensure = "Present",
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Testing Team guest settings for $TeamName"
+    $CurrentValues = Get-TargetResource @PSBoundParameters
+
+    return Test-Office365DSCParameterState -CurrentValues $CurrentValues `
+        -DesiredValues $PSBoundParameters `
+        -ValuesToCheck @("AllowCreateUpdateChannels", `
+            "AllowDeleteChannels", `
+            "Ensure"
+    )
+}
+
+function Export-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.String])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $TeamName,
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+    Test-TeamsServiceConnection -GlobalAdminAccount $GlobalAdminAccount
+    $result = Get-TargetResource @PSBoundParameters
+    $result.GlobalAdminAccount = Resolve-Credentials -UserName $GlobalAdminAccount.UserName
+    $content = "        TeamsGuestSettings " + (New-GUID).ToString() + "`r`n"
```

Comment:
Make sure you also add the logic block to extract those in the Export-O365Configuration method in the Office365DSC module

Reply:
Same as comment below please review added code